bgpd/rfapi/bgp_rfapi_cfg.c

```diff
@@ -3864,491 +3864,469 @@ int bgp_rfapi_cfg_write(struct vty *vty, struct bgp *bgp)
+			vty_out(vty, "  label %u\n", rfg->label);
+		}
+		if (CHECK_FLAG(rfg->flags, RFAPI_RFG_VPN_NH_SELF)) {
+			vty_out(vty, "  nexthop self\n");
+
+		} else {
+			if (rfg->vn_prefix.family) {
+				char buf[BUFSIZ];
+				buf[0] = buf[BUFSIZ - 1] = 0;
+				inet_ntop(rfg->vn_prefix.family,
+					  &rfg->vn_prefix.u.prefix, buf,
+					  sizeof(buf));
+				if (!buf[0] || buf[BUFSIZ - 1]) {
+					// vty_out (vty, "nexthop
+					// self\n");
+				} else {
+					vty_out(vty, "  nexthop %s\n",
+						buf);
 				}
 			}
+		}
 
-			if (rfg->rd.prefixlen) {
-				char buf[RD_ADDRSTRLEN];
-
-				if (AF_UNIX == rfg->rd.family) {
+		if (rfg->rd.prefixlen) {
+			char buf[RD_ADDRSTRLEN];
 
-					uint16_t value = 0;
+			if (AF_UNIX == rfg->rd.family) {
+				uint16_t value = 0;
 
-					value = ((rfg->rd.val[6] << 8)
-						 & 0x0ff00)
-						| (rfg->rd.val[7] & 0x0ff);
+				value = ((rfg->rd.val[6] << 8) & 0x0ff00)
+					| (rfg->rd.val[7] & 0x0ff);
 
-					vty_out(vty, "  rd auto:nh:%d\n",
-						value);
+				vty_out(vty, "  rd auto:nh:%d\n",
+					value);
 
-				} else
-					vty_out(vty, "  rd %s\n",
-						prefix_rd2str(&rfg->rd, buf,
-							      sizeof(buf)));
-			}
+			} else
+				vty_out(vty, "  rd %s\n",
+					prefix_rd2str(&rfg->rd, buf,
+						      sizeof(buf)));
+		}
 
-			if (rfg->rt_import_list && rfg->rt_export_list
-			    && ecommunity_cmp(rfg->rt_import_list,
-					      rfg->rt_export_list)) {
+		if (rfg->rt_import_list && rfg->rt_export_list
+		    && ecommunity_cmp(rfg->rt_import_list,
+				      rfg->rt_export_list)) {
+			char *b = ecommunity_ecom2str(
+				rfg->rt_import_list,
+				ECOMMUNITY_FORMAT_ROUTE_MAP,
+				ECOMMUNITY_ROUTE_TARGET);
+			vty_out(vty, "  rt both %s\n", b);
+			XFREE(MTYPE_ECOMMUNITY_STR, b);
+		} else {
+			if (rfg->rt_import_list) {
 				char *b = ecommunity_ecom2str(
 					rfg->rt_import_list,
 					ECOMMUNITY_FORMAT_ROUTE_MAP,
 					ECOMMUNITY_ROUTE_TARGET);
-				vty_out(vty, "  rt both %s\n", b);
+				vty_out(vty, "  rt import %s\n", b);
 				XFREE(MTYPE_ECOMMUNITY_STR, b);
-			} else {
-				if (rfg->rt_import_list) {
-					char *b = ecommunity_ecom2str(
-						rfg->rt_import_list,
-						ECOMMUNITY_FORMAT_ROUTE_MAP,
-						ECOMMUNITY_ROUTE_TARGET);
-					vty_out(vty, "  rt import %s\n", b);
-					XFREE(MTYPE_ECOMMUNITY_STR, b);
-				}
-				if (rfg->rt_export_list) {
-					char *b = ecommunity_ecom2str(
-						rfg->rt_export_list,
-						ECOMMUNITY_FORMAT_ROUTE_MAP,
-						ECOMMUNITY_ROUTE_TARGET);
-					vty_out(vty, "  rt export %s\n", b);
-					XFREE(MTYPE_ECOMMUNITY_STR, b);
-				}
 			}
-
-			/*
-			 * route filtering: prefix-lists and route-maps
-			 */
-			for (afi = AFI_IP; afi < AFI_MAX; ++afi) {
-
-				const char *afistr =
-					(afi == AFI_IP) ? "ipv4" : "ipv6";
-
-				if (rfg->plist_export_bgp_name[afi]) {
-					vty_out(vty,
-						"  export %s%s prefix-list %s\n",
-						(rfg->type == RFAPI_GROUP_CFG_VRF ? "" : "bgp "),
-						afistr,
-						rfg->plist_export_bgp_name
-							[afi]);
```

Comment:
I'm not a fan of line breaks before indexes -- better to remove whitespace on the indent than to add white space that has semantic implications.

Reply:
You're commenting on a removed hunk

Reply:
yup, did the review off line and added the comment in wrong place. should have been on lines below (3974)